### ISSUE TITLE
Tree generator - Frequent suboptimal solutions

### ISSUE THREAD

**Issue by MLanghof** [EmmittJ/PoESkillTree#339]
For some reason, the tree generator tends to not do well in this case:
![Input](http://image.prntscr.com/image/3501007df17447a5972687e6ea0f3ebf.png)
(Settings: Wand + Shield, Spell + Area. No pseudo attributes, no tagged nodes behind settings window.)
Argh, I just noticed I missed the tagged mana node (see below).

I tend to get something like this:
![Output](http://i.imgur.com/N5rlnah.jpg)

Clearly, removing two 6% es nodes in the Scion wheel and instead spending them to grab Nullification (5% + 10%) would be preferable.
The same thing keeps happening for more achievable ES values (270%, 200%).

It seems that the solver is simply a bit hasty (not enough iterations/population?), since it does sometimes find better/optimal solutions. I've gotten these results even with 5 iterations though, so maybe something more fundamental is wrong...

**Comment by brather1ng** [comment]
Hm, with 5 iterations a result like yours is very rare. I think this issue is just the solver not being good enough in general to reliably handle this case, not something fundemantal.

Also, I did play around with population size/generation count, increasing either doesn't noticeably improve results (which is why I implemented iterations). While that's hard to tell with how random the solver is, it at least wouldn't produce significantly better results (even with really high values; I think there's a cap for both at which increasing them doesn't really do anything).
